src/service/topics.ts

```diff
@@ -131,7 +134,29 @@ const getLatestTopicPost = async (topicName: string): Promise<TopicLatestPostDat
 const verifyCommunityName = async (name) => {
   try {
     const resApi = await api.get(`/topics/is-exist?name=${name}`);
-    console.warn('resApi', JSON.stringify(resApi.data));
+    return resApi.data;
+  } catch (error) {
+    crashlytics().recordError(new Error(error));
+    return error.response.data;
+  }
+};
+
+const submitCommunityName = async (name) => {
+  try {
+    const resApi = await api.post('/topics/create', {name});
+    return resApi.data;
+  } catch (error) {
+    crashlytics().recordError(new Error(error));
+    return error.response.data;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with error handling in the `verifyCommunityName` and `submitCommunityName` functions. If an error occurs, you are returning `error.response.data`. However, if the error is not from the HTTP request (for example, network issues), `error.response` might be undefined, leading to a new error. It would be better to check if `error.response` exists before accessing its `data` property.

```diff
  } catch (error) {
    crashlytics().recordError(new Error(error));
-140:     return error.response.data;
+140:     return error.response ? error.response.data : { message: 'Unexpected error occurred' };
  }
};

const submitCommunityName = async (name) => {
  try {
    const resApi = await api.post('/topics/create', {name});
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
-150:     return error.response.data;
+150:     return error.response ? error.response.data : { message: 'Unexpected error occurred' };
  }
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->